File: Java-Fundamentals-main/08 - Array 2/Insertion Sort.java
public class Solution {  

    public static void insertionSort(int[] arr) {
    	//Your code goes here
        
        int n = arr.length;
        
        for(int i=1;i<n;i++)
        {
            int j = i-1;
            int tmp = arr[i];
            
            while(j>=0 && arr[j]>tmp)
            {
                arr[j+1] = arr[j];
                j--;
            }
            arr[j+1] = tmp;
        }
        
    }

}

File: Popular interview DSA questions in C++/Insertion Sort.cpp
// https://www.codingninjas.com/codestudio/problems/insertion-sort_3155179



#include <bits/stdc++.h> 
void insertionSort(int n, vector<int> &arr){
    // Write your code here.
    for(int i=1;i<n;i++){
        int temp = arr[i];
        int j = i - 1;
        while(temp < arr[j] && j >=0){
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = temp;
    }
}